src/OpcUaStackServer/ServiceSet/MonitorManager.cpp

```diff
@@ -582,8 +582,9 @@ namespace OpcUaStackServer
 		}
 
 		// check monitored item list
+		monitoredItemIds_.erase(it1);
```

Comment:
ok, The ID is not deleted if only one ID exists in the list.

Reply:
Yes, if there are other IDs, `newMonitoredItemIds' is not empty and I insert the rest IDs in L587.